jupyterlab_sql_editor/ipython_magic/trino/trino.py

```diff
@@ -196,6 +209,47 @@ def display_results(self, results, columns, output, limit=20, show_nonprinting=F
             warnings = []
             json_string = pd.DataFrame.from_records(results, columns=columns).to_json(orient="records")
             json_dict = json.loads(json_string)
+            # cast unsafe ints to str for display
+            for row in json_dict:
+                json_array.append(cast_unsafe_ints_to_str(row, warnings))
+            if show_nonprinting:
+                recursive_escape(json_array)
+            display(warnings, JSON(json_array, expanded=args.expand))
+        elif output == "json_cmt":
+            json_array = []
+            warnings = []
+            json_string = pd.DataFrame.from_records(results, columns=columns).to_json(orient="records")
+            json_dict = json.loads(json_string)
+
+            # Returns this list which should contain the dictionaries (rows) of the query
+            my_dictionary_list = []
+            # Iterate through the rows returned (json_dict {list}) and process the columns
+            for row_index, dictionary_entries in enumerate(json_dict):
+                # Goes through each column for a given row, checks the type, and processes them into a dictionary
+                current_row_as_dictionary = {}
+                for (
+                    column_index,
+                    top_level_column,
+                ) in enumerate(dictionary_entries.keys()):
+                    # results[row_index][column_index] returns the row and current column's values. The key or column name
+                    # is found in the top_level_column value within the loop
+                    # We are essentially pairing up the values from the raw "results" variable passed to us with the column
+                    # names that we can extract from the pandas dataframe
+                    if type(results[row_index][column_index]) is list:
+                        current_row_as_dictionary[top_level_column] = [
+                            self.__convert_dictionary(element) for element in results[row_index][column_index]
+                        ]
+                    elif type(results[row_index][column_index]) is trino.client.NamedRowTuple:
+                        current_row_as_dictionary[top_level_column] = self.__convert_dictionary(
+                            results[row_index][column_index]
+                        )
+                    else:
+                        current_row_as_dictionary[top_level_column] = results[row_index][column_index]
```

Comment:
Isn't this just repeating the logic from `__convert_dictionary`? Why not simplify this whole block by using
```suggestion
                    current_row_as_dictionary[top_level_column] = self.__convert_dictionary(results[row_index][column_index])
```